### DISCUSSION TITLE
How to use css-variables, styled-components for theming in Next.js

### DISCUSSION THREAD

**Discussion by rahul3103** [vercel/next.js#20004] [Help]
Right now I have created two themes on body. and its working well on client side.

`global.scss`
```
body[data-theme='light'] {
  --colors-base: #fcfcfc;
  --colors-white: #fff;
}

body[data-theme='dark'] {
  --colors-base: #131417;
  --colors-white: #202022;
```

I have imported `global.scss`  in my `_app.js` and used these variables throughout my project and its working great on client side. But, from server side there is no theming at all. These variables are not getting replaced by color values which when downloads on screen shows a flash of HTML skeleton and then loads the css colors.

**Comment by adebiyial** [comment]
Hi there @rahul3103, is it possible you share some re-created code so we can debug it together?

You can also check out the [Comparing Styling Methods in Next.js](https://www.smashingmagazine.com/2020/09/comparison-styling-methods-next-js/) article.

**Reply by rahul3103** [reply]
Yeah sure, I will share code sandbox link.

**Reply by rahul3103** [reply]
<img width="1042" alt="Screenshot 2020-12-09 at 2 58 18 PM" src="https://user-images.githubusercontent.com/10813252/101611080-0ebbe180-3a2f-11eb-8502-6fbcbf4521ce.png">

[Link to website](https://7gu2e.sse.codesandbox.io/)

You can check from server, i am getting black but on client its becoming red.

https://codesandbox.io/s/quirky-mendeleev-7gu2e

**Reply by rahul3103** [reply]
@adebiyial Can we debug this together, whenever you are free?

**Comment by BassOfBass** [comment]
Does it flash if css-variables are set in `.css` file?

**Reply by rahul3103** [reply]
yes

**Comment by gupta-ji6** [comment]
@rahul3103 In my experience, yes it will always flash.

Since the applied value (theme) will be known only on the client side, it will flash if the set value is different from the default value.

To avoid flashing, you can use this library - [next-themes](https://github.com/pacocoursey/next-themes). It works like a charm 👌🏻

**Reply by rahul3103** [reply]
But, if you see how github they are doing, they are using css-variables
